src/Functions/dateDiff.cpp

```diff
@@ -34,6 +34,163 @@ namespace ErrorCodes
 namespace
 {
 
+class DateDiffImpl
+{
+using ColumnDateTime64 = ColumnDecimal<DateTime64>;
+public:
+explicit DateDiffImpl(const String& name_) : name(name_) {}
```

Comment:
indentation is wrong.

Also maybe it would be cleaner to pass a function name as an argument (but using base class is also ok to me).